### ISSUE TITLE
Heroku: 500 Error: failed to retrieve apps list:

### ISSUE THREAD

**Issue by useo6** [Strider-CD/strider#1052]
When I try to add a Heroku account, I'm getting the following error:

> Connect
> 500 Error: failed to retrieve apps list: Status: 400; {"id":"missing_version","error":"Please specify a version along with Heroku's API MIME type. For example, `Accept: application/vnd.heroku+json; version=3`.\n"}
> 
>        at /root/strider/node_modules/strider-heroku/webapp.js:132:26
>        at /root/strider/node_modules/strider-heroku/lib/api.js:14:38
>        at Request.callback (/root/strider/node_modules/strider-heroku/node_modules/superagent/lib/node/index.js:630:3)
>        at Request.<anonymous> (/root/strider/node_modules/strider-heroku/node_modules/superagent/lib/node/index.js:131:10)
>        at emitOne (events.js:116:13)
>        at Request.emit (events.js:211:7)
>        at IncomingMessage.<anonymous> (/root/strider/node_modules/strider-heroku/node_modules/superagent/lib/node/index.js:773:12)
>        at emitNone (events.js:111:20)
>        at IncomingMessage.emit (events.js:208:7)
>        at endReadableNT (_stream_readable.js:1056:12)
>        at _combinedTickCallback (internal/process/next_tick.js:138:11)
>        at process._tickDomainCallback (internal/process/next_tick.js:218:9)

Where do I need to specify the version and API MIME type?

**Comment by xgalen** [comment]
Hi, I don't work with Heroku but maybe you can try with this: https://futurestud.io/tutorials/strider-continuous-deployment-to-heroku

**Comment by Norbz** [comment]
Getting the exact same error just after giving authorization to heroku.

**Comment by knownasilya** [comment]
Try updating to v0.1.3 of the heroku plugin.

**Comment by Norbz** [comment]
Just updated the plugin to 0.1.3. I'm getting a new error now :

```
00 Error: Failed to add ssh key: Status: 404; {"id":"not_found","message":"The requested API endpoint was not found. Are you using the right HTTP verb (i.e. `GET` vs. `POST`), and did you specify your intended version with the `Accept` header?"}
   at /home/strider/strider/node_modules/strider-heroku/webapp.js:136:30
   at /home/strider/strider/node_modules/strider-heroku/lib/api.js:36:38
   at Request.callback (/home/strider/strider/node_modules/strider-heroku/node_modules/superagent/lib/node/index.js:630:3)
```

Should I clean anything in the database and start over ?

**Comment by Norbz** [comment]
Cleaned the project, removed the authorisation token, and started everything over. I still have the issue.

I can see my authorisation token for strider in my account, but somehow when requesting my app list, I get this issue.

Is there anything I missed in the configuration ?

**Comment by knownasilya** [comment]
Looks like they changed the whole API without versioning it, so we'll have to figure out how to do keys again. I looked a bit this weekend and didn't find an equivalent to the current endpoint for adding keys. Will have to do more research.

**Comment by Norbz** [comment]
Yes, I was just searching this too. Can't find a new endpoint that would match it.
The heroku-cli is still able to add some keys though, I will have a look in the CLI source-code.

**Comment by knownasilya** [comment]
Thanks, I appreciate it.

**Comment by Norbz** [comment]
Any news on this ? Couldn't find what I wanted from the CLI sources, could we move to another way of adding the key by invoking the CLI ?

**Comment by knownasilya** [comment]
Will try to find a way to do that, would prefer not to run a CLI just to set up the keys 😢

**Comment by creaux** [comment]
Hello guys, any news on this?

**Comment by knownasilya** [comment]
I asked on Twitter but have not put any official support tickets in. There was no response last I remember. If you are interested it would help to put together a support ticket to get instructions on how to add keys in the new API version.

**Comment by creaux** [comment]
@knownasilya can you please provide that twitter message?

**Comment by knownasilya** [comment]
https://twitter.com/knownasilya/status/1017480613585195009

**Comment by creaux** [comment]
I have posted ticket to Heroku. They don't like to share it publicly so I will come back here as soon as I will have a response from them.

**Comment by creaux** [comment]
Heroku response:

> Since our documentation does not reference such an endpoint, I can only assume that while one might exist, it isn't a public endpoint. That is to say that even if you had found something that worked, if it wasn't a supported endpoint, we consider it something that can be changed without consideration for external users.

That said, here's what I can tell you: you were asking about how to specify the version. You would do so in the Accept header of your API call. To cite our API documentation for the Key Info endpoint:

$ curl -n https://api.heroku.com/account/keys/$KEY_ID_OR_FINGERPRINT \
  -H "Accept: application/vnd.heroku+json; version=3"
That said, it may be suggesting that you specify the version because it doesn't recognize the endpoint. Someone in that Github Issue aptly mentioned that we may have changed the endpoints without versioning it. This very well could be the case. We only bump the version when changing the public API endpoints.

Alternatively, we would recommend that you consider using the heroku keys:add command for this when scripting the addition of keys. Hopefully one of these two will help you achieve your goal.

**Comment by knownasilya** [comment]
Hum.. so we do use the version number in the header here https://github.com/Strider-CD/strider-heroku/blob/1cab3de70ffb9def5a2735cfdaf990b858d3a056/lib/api.js#L31

Looks like we'll need to use the cli tool and wrap it for our use-case.

**Comment by knownasilya** [comment]
Fixed in Strider 2.4.5. You'll need to use `heroku login` before strider can add ssh keys
